dhis-2/dhis-services/dhis-service-data-exchange/src/main/java/org/hisp/dhis/dataexchange/aggregate/AggregateDataExchangeJob.java

```diff
@@ -65,6 +69,11 @@ public void execute(JobConfiguration config, JobProgress progress) {
     AggregateDataExchangeJobParameters params =
         (AggregateDataExchangeJobParameters) config.getJobParameters();
 
+    UserDetails currentUser =
+        config.getExecutedBy() != null
+            ? UserDetails.fromUser(userService.getUser(config.getExecutedBy()))
+            : CurrentUserUtil.getCurrentUserDetails();
```

Comment:
Should be just `CurrentUserUtil.getCurrentUserDetails()` - if the `executedBy` user is set at job construction, that user will be the current user when running the `execute` method of the job. 

Note that if no user is set at construction this is the `SuperUser`!

Reply:
Done